### PR TITLE
chore(menu): fixes after testing described in the related item

### PR DESCRIPTION
**Pull request by mvgmvg**
#83
#78

### CODE REVIEW

File: packages/default/scss/menu/_layout.scss [hunk 1]
```diff
@@ -91,41 +94,23 @@
...
-    .k-animation-container .k-context-menu.k-menu-horizontal {
-        display: flex !important; // sass-lint:disable-line no-important
-    }
-
-
     // Orientation
     .k-menu-horizontal {
 
-        > .k-item + .k-item {
+        > .k-item + .k-item:not(.k-separator) {
```
Comment (gyoshev) [inline packages/default/scss/menu/_layout.scss]: Instead of making the item selector more specific, use `.k-item.k-separator`. This will reduce the need to add more selectors to undo the changes that apply due to the high specificity.

File: packages/material/scss/menu/_layout.scss [hunk 2]
```diff
@@ -60,8 +61,11 @@ $item-selected-opacity: .16 !default;
     .k-menu-scroll-wrapper {
 
         .k-menu-scroll-button {
-            padding: $button-padding-x;
             box-shadow: none;
         }
+
+        &:not(.vertical) .k-menu-scroll-button {
```
Comment (gyoshev) [inline packages/material/scss/menu/_layout.scss]: I was puzzled why this rule applies, given the `vertical` class name. I believe this class should be renamed by convention.

**Comment by mvgmvg** [comment]
The new spacing between item icon and its text is uniformed for all items (root and sub-items), for sass themes. The new value is 8 pixels. ( consulted with @lkarakoleva )

File: packages/default/scss/menu/_layout.scss [hunk 3]
```diff
@@ -1,7 +1,10 @@
+$menu-separator-spacer: 2px !default;
+
 @include exports("menu/layout") {
 
     // Base
-    .k-menu {
+    .k-menu,
+    .k-menu-scroll-wrapper {
```
[the remark this answers is not in the document]
Comment (mvgmvg) [reply packages/default/scss/menu/_layout.scss]: Yes, it can. Every sub-menu in a scrollable menu is placed inside the k-menu-scroll-wrapper.

File: packages/material/scss/menu/_layout.scss (hunk 2 above)
Comment (mvgmvg) [reply packages/material/scss/menu/_layout.scss]: I have logged [an issue](https://github.com/telerik/kendo/issues/8392) to fix the class name by convention. For this moment this class is used also in LESS themes.

File: packages/default/scss/menu/_layout.scss [hunk 4]
```diff
@@ -145,10 +130,51 @@
...
+        &.k-menu-horizontal {
+            padding: $popup-padding-x $popup-padding-y;
+        }
+
+        & > .k-item + .k-item:not(.k-separator) {
+            margin-left: 0;
+        }
+    }
+    .k-animation-container .k-context-menu.k-menu-horizontal {
+        display: flex !important; // sass-lint:disable-line no-important
```
[the remark this answers is not in the document]
Comment (joneff) [reply packages/default/scss/menu/_layout.scss]: The reason for the `!important` is overriding inline styles, namely `display: block`. I keep thinking that we should refrain from using `display: none; display: block;` combo from javascript and use `.k-hidden` instead...

I will update the code momentarily

File: packages/default/scss/menu/_layout.scss (hunk 1 above)
Comment (mvgmvg) [reply packages/default/scss/menu/_layout.scss]: Separator selectors are improved

File: packages/default/scss/menu/_layout.scss [hunk 5]
```diff
@@ -202,12 +219,19 @@
 
 
     // RTL
-    .k-rtl {
+
+    .k-rtl,
+    [dir="rtl"] {
+
+        .k-menu:not(.k-menu-vertical) {
+            flex-direction: row-reverse;
```
Comment (gyoshev) [inline packages/default/scss/menu/_layout.scss]: Why is this rule necessary? Flexbox containers automatically reverse the horizontal direction when in an RTL container ([see Dojo](http://dojo.telerik.com/ESUMitiT)). The horizontal RTL menu needs to be laid out in reverse order, and this rule prohibits that.

File: packages/default/scss/menu/_layout.scss [hunk 6]
```diff
@@ -145,10 +130,51 @@
...
             position: relative;
+
+            & > .k-image,
+            & > .k-sprite {
+                margin-right: $padding-x;
+
+                .k-rtl &,
+                [dir="rtl"] & {
+                    margin-left: $padding-x;
+                    margin-right: 0;
```
[the remark this answers is not in the document]
Comment (mvgmvg) [reply packages/default/scss/menu/_layout.scss]: the issue is fixed

File: packages/default/scss/menu/_layout.scss [hunk 7]
```diff
@@ -1,10 +1,17 @@
...
 
     // Base
     .k-menu {
         display: flex;
         flex-direction: row;
         align-items: stretch;
+    }
+
+    .k-menu,
+    .k-menu-scroll-wrapper {
```
[the remark this answers is not in the document]
Comment (mvgmvg) [reply packages/default/scss/menu/_layout.scss]: Styles placed inside this selector are used also for sub-items. When a menu is scrollable, its submenu container is not a child of k-menu. The animation container with sub-items is a sibling of the .k-menu.k-widget wrapper, placed inside the k-menu-scroll-wrapper. To prevent redundant rules, we can use ".k-menu-scroll-wrapper > .k-menu-group" selector instead of ".k-menu-scroll-wrapper" only. But in this way, the selector will use more class names and it will be necessary to use a stronger selector for overwriting these rules.

[the remark this answers is not in the document]
Comment (mvgmvg) [reply packages/default/scss/menu/_layout.scss]: I have added a new commit, with another option, which uses menu-group selector approach instead. https://github.com/telerik/kendo-themes/pull/127/commits/7100c2c2f3adc84f8e1c924bba43700c4c0e0a46
